### ISSUE TITLE
Content being replaced by another language, in between reverting to revision, saving a draft and changing languages

### ISSUE THREAD

**Issue by hhentschel** [craftcms/cms#10206]
### Description
We have a multisite (4 languages) and we get feedback from the customer, 
that content in elements (made via neo blocks) from one language disappears
and is replaced by the another language, when switching between revisions, drafts and languages.
Customer cannot tell us the exact steps, but it happens at least once a week and they are working on articles
every day. 
We tried to reproduce it ourselves for a couple of hours within 2 days and it happened to us two times.
But as same as the customers tells us, it happens "suddenly" and we are not able to reproduce it right afterwards.
We assume it must be connected to revisions and drafts.

### Steps to reproduce
Can't really reproduce, it "just happens suddenly" once in a while.
Must be somewhere in between setting a draft, reverting to a revision and applying the draft.
1.
2.

### Additional info

- Craft version: 3.7.20
- PHP version: 7.4.25
- Database driver & version: MySQL 5.5.5
- Plugins & versions: 

Expanded Singles | 1.1.4
Neo | 2.11.16
String Urldecode | 1.0.0
Redactor | 2.8.8
SEOmatic | 3.4.17

**Comment by brandonkelly** [comment]
This sounds similar to #10106, which was fixed in 3.7.22. Please update Craft to the latest version and see if the problem persists.

**Comment by hhentschel** [comment]
We updated to 3.7.22 and customer reports that the issue still exists.

**Comment by brandonkelly** [comment]
Alright, well we are going to need specific steps to reproduce reliably. If you can figure that out, that will give us something to go on.

**Comment by hhentschel** [comment]
It seems to be an hard one.
This is what i got back from the customer, it happened again yesterday:
---
Each page is filled with the content in the appropriate language, i.e. German, French and Italian. After that, all the pages are displayed correctly on the website with all the translations filled in.
-       Then, for example, a text module is adjusted in a content element on the German page.
-       Next, the “Save” button is pressed to apply the changes.
-       And this is exactly where the problem occurs: If you then want to adapt the corresponding text module in the other two languages French and Italian as well, it has already happened several times that all pages are suddenly displayed in German only.
---
It basically mean, after saving an entry, the translations are not stored correctly. I tried it on our customers dev environment, and could not reproduced it. I did reproduced it last week, but as stated before, somewhere in between translation and revisions change. (2 times within an hour of changing and saving...)

I know, its not much to work with. is there something we can do to find a solution?
sending the `composer.json` and `composer.lock`?
db is quite big (189MB).
